lib/mayaUsd/ufe/UsdContextOps.cpp

```diff
@@ -425,6 +461,72 @@ _computeLoadAndUnloadItems(const UsdPrim& prim)
 namespace MAYAUSD_NS_DEF {
 namespace ufe {
 
+std::vector<SchemaTypeGroup> UsdContextOps::fSchemaTypeGroups = {};
+
+bool _schemaGroupSortCompare(const SchemaTypeGroup& a, const SchemaTypeGroup& b)
+{
+    return a.name < b.name;
+}
+
+//! \brief Get groups of concrete schema prim types to list dynamically in the UI
+static const std::vector<SchemaTypeGroup> getConcretePrimTypes(bool sorted)
+{
+    std::vector<SchemaTypeGroup> groups;
+
+    // Query all the available types
+    PlugRegistry&    plugReg = PlugRegistry::GetInstance();
+    std::set<TfType> schemaTypes;
+    plugReg.GetAllDerivedTypes<UsdSchemaBase>(&schemaTypes);
+
+    for (auto t : schemaTypes) {
+        if (!UsdSchemaRegistry::IsConcrete(t)) {
+            continue;
+        }
+
+        auto plugin = plugReg.GetPluginForType(t);
+        if (!plugin) {
+            continue;
+        }
+
+        // For every plugin we check if there's a nice name registered and use that instead
+        auto plugin_name = plugin->GetName();
+        auto name_itr
+            = std::find(kSchemaPluginNames.begin(), kSchemaPluginNames.end(), plugin_name);
+        if (name_itr != kSchemaPluginNames.end()) {
+            plugin_name = kSchemaNiceNames[name_itr - kSchemaPluginNames.begin()];
+        }
+
+        // We don't list empty names. This allows hiding certain plugins too.
+        if (plugin_name.empty()) {
+            continue;
+        }
+
+        auto type_name = UsdSchemaRegistry::GetConcreteSchemaTypeName(t);
+
+        // Find or create the schema group and add to it
+        auto group_itr = find(begin(groups), end(groups), plugin_name);
+        if (group_itr == groups.end()) {
+            SchemaTypeGroup group { plugin_name };
+            group.types.emplace_back(type_name);
+            groups.emplace_back(group);
+        } else {
+            groups[group_itr - groups.begin()].types.emplace_back(type_name);
+        }
+    }
+
+    if (sorted) {
+        for (size_t i = 0; i < groups.size(); ++i) {
+            auto group = groups[i];
+            std::sort(group.types.begin(), group.types.end());
+            groups[i] = group;
+        }
+
+        std::sort(groups.begin(), groups.end(), _schemaGroupSortCompare);
+    }
+
+    return groups;
+}
```

Comment:
All of this code should be moved outside of the this MayaUsd::Ufe namespace and into the unnamed namepsace that ends just above on line 459.